### PR TITLE
feat: Add sync B2B validations to wholesale serives requests

### PR DESCRIPTION
**Pull request by MWO1024**

### CODE REVIEW

File: source/IncomingMessages.Infrastructure/Messages/RequestAggregatedMeasureData/MessageTypeValidator.cs [hunk 1]
```diff
@@ -12,17 +12,34 @@
...
+
+        return await Task.FromResult(
+                message switch
+                {
+                    RequestAggregatedMeasureDataMessage ramdm =>
+                        _aggregatedMeasureDataWhiteList.Contains(ramdm.MessageType)
+                            ? Result.Succeeded()
+                            : Result.Failure(
+                                new NotSupportedMessageType(ramdm.MessageType)),
+                    RequestWholesaleServicesMessage rqsm =>
```
Comment (RasmusGraabaek) [inline source/IncomingMessages.Infrastructure/Messages/RequestAggregatedMeasureData/MessageTypeValidator.cs:37]:
```suggestion
                    RequestWholesaleServicesMessage rwsm =>
```
?

File: source/IncomingMessages.Infrastructure/Messages/RequestAggregatedMeasureData/ProcessTypeValidator.cs [hunk 2]
```diff
@@ -12,17 +12,34 @@
...
 
 public class ProcessTypeValidator : IProcessTypeValidator
 {
-    private static readonly IReadOnlyCollection<string> _whiteList = new[] { "D03", "D04", "D05", "D32" };
+    private static readonly IReadOnlyCollection<string> _aggregatedMeasureDataWhitelist =
+        new[] { "D03", "D04", "D05", "D32" };
 
-    public async Task<Result> ValidateAsync(string processType, CancellationToken cancellationToken)
+    private static readonly IReadOnlyCollection<string> _wholesaleServicesWhitelist =
+        new[] { "D05", "D32" };
```
Comment (RasmusGraabaek) [inline source/IncomingMessages.Infrastructure/Messages/RequestAggregatedMeasureData/ProcessTypeValidator.cs:26]: Should we use `BusinessReason.WholesaleFixing.Code` here?

File: source/IncomingMessages.Infrastructure/Messages/RequestAggregatedMeasureData/ProcessTypeValidator.cs [hunk 3]
```diff
@@ -12,17 +12,34 @@
...
+
+    public async Task<Result> ValidateAsync(IIncomingMessage message, CancellationToken cancellationToken)
     {
-        return await Task.FromResult(_whiteList.Contains(processType) ?
-             Result.Succeeded() : Result.Failure(new NotSupportedProcessType(processType))).ConfigureAwait(false);
+        return await Task.FromResult(
+                message switch
+                {
+                    RequestAggregatedMeasureDataMessage ragdm =>
```
Comment (RasmusGraabaek) [inline source/IncomingMessages.Infrastructure/Messages/RequestAggregatedMeasureData/ProcessTypeValidator.cs:33]:
```suggestion
                    RequestAggregatedMeasureDataMessage ramdm =>
```
That's the naming in "source/IncomingMessages.Infrastructure/Messages/RequestAggregatedMeasureData/MessageTypeValidator.cs"

File: source/IntegrationTests/Application/IncomingMessages/RequestWholesaleSettlementWithUnusedBusinessReason.json [hunk 4]
```diff
@@ -13,7 +13,7 @@
 			"value": "5790001330552"
 		},
 		"receiver_MarketParticipant.marketRole.type": {
-			"value": "DDZ"
+      "value": "DGL"
```
Comment (RasmusGraabaek) [inline source/IntegrationTests/Application/IncomingMessages/RequestWholesaleSettlementWithUnusedBusinessReason.json:16]: Does this file have any references?

**Review by RasmusGraabaek — APPROVED** [review]

File: source/IncomingMessages.Infrastructure/Messages/RequestAggregatedMeasureData/ProcessTypeValidator.cs (hunk 2 above)
Comment (MWO1024) [reply source/IncomingMessages.Infrastructure/Messages/RequestAggregatedMeasureData/ProcessTypeValidator.cs:26]: I think so. Good catch!

File: source/IntegrationTests/Application/IncomingMessages/RequestWholesaleSettlementWithUnusedBusinessReason.json (hunk 4 above)
Comment (MWO1024) [reply source/IntegrationTests/Application/IncomingMessages/RequestWholesaleSettlementWithUnusedBusinessReason.json:16]: Not anymore I think. At least I can't find a reference. I could add it to `InvalidIncomingRequestMessages` in `WhenIncomingMessagesIsReceivedTests`?

Comment (MWO1024) [reply source/IntegrationTests/Application/IncomingMessages/RequestWholesaleSettlementWithUnusedBusinessReason.json:16]: I decided to add it. We can remove it later.

**Review by RasmusGraabaek — APPROVED** [review]

**Pull request merged**